client/client.go

```diff
@@ -762,6 +763,21 @@ func (c *Client) init() error {
 	// setup the nsd check store
 	c.checkStore = checkstore.NewStore(c.logger, c.stateDB)
 
+	// COMPAT(1.12.0): remove in Nomad 1.12.0
+	oldCNIDir := "/var/lib/cni/networks/nomad"
+	newCNIDir := "/var/run/cni/nomad"
+	if _, err := os.Stat(newCNIDir); os.IsNotExist(err) {
```

Comment:
this assumes that if the new dir exists at all, then no migration is necessary, which I think is okay? thinking through it:

1. old agent running with networks in the old dir
2. upgrade to newer agent with new dir, but not with this fix
3. restoring allocs creates fresh networks?
   * no, only new allocs get networks in the new dir, right?
5. and leaves both old and new dir on disk
6. upgrade to agent with this code - it uses the new dir but migrates nothing

I suppose the new dir in step 2 has become the new source of truth for new allocs (causing the bug by being unaware of old networks), and we can't reasonably un-hit that bug, so no use to try and restore the old state.  is that right?

I'm having some trouble mapping it out mentally, but something smells funny.

Reply:
Yeah there are really two cases where we'd see the new dir already:
* We're restarting the client  agent after we've already upgraded and migrated happily. Here we don't want to do anything.
* We're upgrading the client from the broken version to the new version. Here we've already broken all our old allocs, and any new allocs will be writing to the new dir already, so there's no point in trying to copy the old state because we're just going to make things worse.

Reply:
> We should be removing the oldCniDir once successfully migrated to not keep triggering the migration which would happen after reboot. In that case /var/run will be empty and we’d be copying back the previous kept state?

I went back and forth on that, because I was worried about trying to downgrade a client. But we don't really support that and it makes things easier if we can get rid of it. Done.